catalyst_voices/apps/voices/lib/widgets/rich_text/placeholder_rich_text.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter/material.dart';
+
+final _kPlaceholderRegExp = RegExp(r'{(\w*)}');
+
+typedef PlaceholderSpanBuilder = InlineSpan Function(
+  BuildContext context,
+  String placeholder,
+);
+
+class PlaceholderRichText extends StatefulWidget {
+  const PlaceholderRichText(
+    this.text, {
+    super.key,
+    required this.placeholderSpanBuilder,
+    this.style,
+    this.textAlign = TextAlign.start,
+  });
+
+  final String text;
+  final PlaceholderSpanBuilder placeholderSpanBuilder;
+  final TextStyle? style;
+  final TextAlign textAlign;
```

Comment:
FIelds first, constructor below

```suggestion
  final String text;
  final PlaceholderSpanBuilder placeholderSpanBuilder;
  final TextStyle? style;
  final TextAlign textAlign;

  const PlaceholderRichText(
    this.text, {
    super.key,
    required this.placeholderSpanBuilder,
    this.style,
    this.textAlign = TextAlign.start,
  });
```